python/plugins/processing/algs/qgis/CreateAtlasGrid.py

```diff
@@ -0,0 +1,178 @@
+
+from processing.algs.qgis.QgisAlgorithm import QgisAlgorithm
+
+from qgis.PyQt.QtCore import QVariant
+from qgis.core import (QgsRectangle,
+                       QgsProject,
+                       QgsProcessingException,
+                       QgsProcessingParameterString,
+                       QgsProcessingParameterFeatureSource,
+                       QgsProcessingParameterFeatureSink,
+                       QgsProcessingParameterBoolean,
+                       QgsProcessingParameterDistance,
+                       QgsProcessing,
+                       QgsFields,
+                       QgsField,
+                       QgsWkbTypes,
+                       QgsFeature,
+                       QgsGeometry)
+
+
+class CreateAtlasGrid(QgisAlgorithm):
+
+    LAYOUT = 'LAYOUT'
+    COVERAGE_LAYER = 'COVERAGE_LAYER'
+    COVERAGE_BUFFER = 'COVERAGE_BUFFER'
+    ONLY_ON_FEATURES = 'ONLY_ON_FEATURES'
+    OUTPUT = 'OUTPUT'
+
+    def name(self):
+        # Unique (non-user visible) name of algorithm
+        return 'createatlasgrid'
+
+    def displayName(self):
+        # The name that the user will see in the toolbox
+        return self.tr('Create a grid from map layout')
+
+    def group(self):
+        return self.tr('Cartography')
+
+    def groupId(self):
+        return 'cartography'
+
+    def __init__(self):
+        super().__init__()
+
+    def initAlgorithm(self, config=None):
+
+        layout_param = QgsProcessingParameterString(
+            self.LAYOUT,
+            description=self.tr('Layout to use'),
+            defaultValue=None,
+            optional=False)
+
+        layout_param.setMetadata(
+            {'widget_wrapper': {
+                'class':
+                    'processing.gui.wrappers_layout.LayoutWrapper'}})
+        self.addParameter(layout_param)
+
+        self.addParameter(QgsProcessingParameterFeatureSource(self.COVERAGE_LAYER,
+                                                              self.tr('Coverage layer'),
+                                                              [QgsProcessing.TypeMapLayer]))
+        self.addParameter(QgsProcessingParameterDistance(self.COVERAGE_BUFFER,
+                                                         self.tr('Coverage Buffer'), parentParameterName='COVERAGE_LAYER',
+                                                         defaultValue=0.0))
+        self.addParameter(QgsProcessingParameterBoolean(self.ONLY_ON_FEATURES,
+                                                        self.tr('Only on intersection features'),
+                                                        defaultValue=True))
+        self.addParameter(QgsProcessingParameterFeatureSink(self.OUTPUT,
+                                                            self.tr('Grid'),
+                                                            QgsProcessing.TypeVectorPolygon))
+
+    def processAlgorithm(self, parameters, context, feedback):
+
+        layout_name = self.parameterAsString(
```

Comment:
use parameterAsEnum if the parameter is used is an Enum (see comment above)

Reply:
I use parameterAsString with a wrapper that returns all print layouts that hace a referenceMap setted. The gui control is showed as comboBox and don't alow users to add custom values. I use this solution based on other algs.

Reply:
wait for Nyall opinion... my suggestio is just to avoid to use a wrapper (e.g. more code) whan it is possible to use less (less errors)